request.js

```diff
@@ -1413,6 +1413,28 @@ Request.prototype.jar = function (jar) {
   return self
 }
 
+Request.prototype.cloneJar = function (cookieJar) {
+  var self = this
+  var newJar, cookieString, cookieStore, domains, domainPaths, cookieUrl
+
+  newJar = self.jar()
+  cookieStore = cookieJar._jar.store.idx
```

Comment:
Looking more into this it seems that relying on `idx` is brittle.
`request.jar` takes in a store argument, and that gets passed all the way down to tough-cookie's CookieJar.
That doesn't rely on `idx` directly, but an API on the store. Meaning that it's not required to have the property `idx` it could be named anything.